modules/standard/Regex.chpl

```diff
@@ -1153,28 +1153,52 @@ inline operator :(x: bytes, type t: regex(bytes)) throws {
   return compile(x);
 }
 
+/* Search the receiving string for a regular expression already compiled
+   by calling :proc:`regex.search`. Search for matches at any offset.
```

Comment:
I think this might go a little too into implementation details.  Maybe: `Search the receiving string for the result of a compiled regular expression.  Search for matches at any offset.` ?

It looks like you got this from other documentation in the file.  I think the same applies to those as well, but it's okay if you don't fix them in this PR.  Since you're adding these paragraphs, I do think it would be nice to avoid the implementation details, though.

Reply:
The "by calling :proc:`regex.search`" part is my main concern, I don't think the user needs to know necessarily that we're calling the regex's search method.